src/ipc/ipc4/handler.c

```diff
@@ -251,6 +251,36 @@ static struct ipc_comp_dev *pipeline_get_host_dev(struct ipc_comp_dev *ppl_icd)
 	struct ipc *ipc = ipc_get();
 	int host_id;
 
+	/**
+	 * @brief Synchronize the direction property between source and sink components.
+	 *
+	 * If the source component's direction is not set but the sink's direction is,
+	 * this block will copy the direction from the sink to the source component and
+	 * mark the source's direction as set. This ensures that the pipeline's flow
+	 * direction is consistent and correctly configured for the traversal process.
+	 */
+	if (!ppl_icd->pipeline->source_comp->direction_set &&
```

Comment:
Identical code and logic is already done in ipc_comp_connect(). It would be interesting to understand better why this can happen in some case. If this cannot be avoided, can we have a shared helped used by both this function and ipc_comp_connect() ?